cmake/CMakeLists.txt

```diff
@@ -2,7 +2,15 @@
 # CMake build system
 # This file is part of LAMMPS
 # Created by Christoph Junghans and Richard Berger
-cmake_minimum_required(VERSION 2.8.12)
+if(PKG_USER-ADIOS)
+  message(STATUS "Force newer standards because of using ADIOS")
+  cmake_minimum_required(VERSION 3.6)
```

Comment:
Just move a 
```cmake
if (CMAKE_VERSION VERSION_LESS "3.6")
  message(FATAL_ERROR "For the USER-ADIOS package you need at least cmake-3.6")
endif()
```
in the `if(USER-ADIOS)` block below.

Also why 3.6 anything specific with that version?